### PR TITLE
FLUID-5818: Fix for failure of ginger reference from constructed child of unconstructed parent

### PR DESCRIPTION
**Pull request by amb26**
New tip of "shadow trunk"

### CODE REVIEW

**Comment by colinbdclark** [comment]
This pull request causes the Builder unit tests to fail consistently on all browsers. Here's the output of the failing test:

Prefs editor with composite panel: Rendering (1, 12, 13)
The single panel's checkbox should be in the correct state - at sequence position 1 of 5
The composite panel should be rendered correctly - at sequence position 1 of 5
Expected: 	
"increase"
Result: 	
"[Message string for key increaseHeader not found]"
Diff: 	
"increase" "[Message string for key increaseHeader not found]" 
Source: 	
    at Object.jsUnitCompat.assertEquals (http://localhost:8888/infusion/tests/test-core/jqUnit/js/jqUnit.js:135:19)
    at fluid.tests.composite.tester.initialRendering (http://localhost:8888/infusion/tests/framework-tests/preferences/js/BuilderTests.js:874:16)
    at wrapped (http://localhost:8888/infusion/tests/test-core/utils/js/IoCTestUtils.js:323:26)
    at Object.that.fire (http://localhost:8888/infusion/src/framework/core/js/Fluid.js:1399:40)
    at Function.fluid.prefs.prefsEditor.finishInit (http://localhost:8888/infusion/src/framework/preferences/js/PrefsEditor.js:454:29)
    at http://localhost:8888/infusion/src/framework/preferences/js/PrefsEditor.js:469:41

**Comment by jobara** [comment]
@colinbdclark and @amb26 I confirmed this issue in FF 43.0.2 on Mac OS and Chrome 47  on Win 8.1

**Comment by amb26** [comment]
@colinbdclark, @jobara - thanks for the testing, fix now pushed here (reported at https://issues.fluidproject.org/browse/FLUID-5835 )

File: src/framework/core/js/Fluid.js [hunk 1]
```diff
@@ -431,15 +431,15 @@ var fluid = fluid || fluid_2_0_0;
...
+    /** Return an array or hash of objects, transformed by one or more functions. Similar to
      * jQuery.map, only will accept an arbitrary list of transformation functions and also
      * works on non-arrays.
      * @param source {Array or Object} The initial container of objects to be transformed. If the source is
      * neither an array nor an object, it will be returned untransformed
      * @param fn1, fn2, etc. {Function} An arbitrary number of optional further arguments,
      * all of type Function, accepting the signature (object, index), where object is the
-     * list member to be transformed, and index is its list index. Each function will be
-     * applied in turn to each list member, which will be replaced by the return value
+     * structure member to be transformed, and index is its key orindex. Each function will be
```
Comment (jobara) [inline src/framework/core/js/Fluid.js]: typo "orindex"

**Comment by amb26** [comment]
Thanks for catching that typo, @jobara - now fixed

**Comment by jobara** [comment]
merged at 4e1b8c9a64665c67232901d2e57531926ec7247f  and 2f38c09d733925441f0ff2e6636f8886da8af5ea

**Pull request merged**